docs-chef-io/content/server/upgrade_server_ha_v2.md

```diff
@@ -163,13 +163,160 @@ Upgrading from Chef Backend 1.x to Chef Backend 2.x requires full cluster downti
-|---------|---------|------|
-| 2.x | 2.x | Rolling |
-| 2.1 | 3.0 | Rolling |
-| 2.0 | 3.0 | Downtime |
+### Planning
+
+The versions of Chef Backend, Elasticsearch, and Chef Infra Server running your system determines the type of upgrade, index treatment, and steps for your upgrade.
+
+#### Upgrade Type
+
+Systems with Elasticsearch 5.4.1 or lower require downtime. Systems with Elasticsearch 5.6.16 or higher use a rolling upgrade.
+
+| Running Backend Version | Upgrade To Version | Elasticsearch Version | Upgrade Type |
+|---------|---------|------|------|
+| 2.1 | 3.0 | 5.4.1 or lower | Downtime |
+| 2.1 | 3.0 | 5.6.16 or greater | Rolling |
+
+Find the version of Elasticsearch on your system with the command:
+
+```bash
+curl -XGET localhost:9200/chef/_settings/index.version.created*?pretty\&human
+```
+
+#### Index Treatment
+
+Chef Backend clusters that were originally installed with 2.3.0 or earlier _and then_ upgraded to version 2.3.1 will need to delete and reindex the Elasticsearch after upgrading. Chef Backend clusters first installed with version 2.3.1 or later will automatically re-index.
+
+To determine if you need to delete and replace the indexes, run the command:
+
+```bash
+curl -XGET localhost:9200/chef/_settings/index.version.created*?pretty\&human
+```
+
+| Original Chef Backend Install | Identifier | Index Treatment |
+|---------|-----------|---------|
+| 2.3.0 or earlier| `"created_string" : "2.3.1"`| Delete and reindex |
+| 2.3.1 or later |`"created_string" : "5.6.16"`| Automatic |
+
+#### Chef Infra Server Version
+
+Find the Chef Infra Server version on your system.
+
+TODO: Is it in the Chef Infra Server `/opt/chef-backend/version-manifest.json`.
+TODO: Does `chef-server-ctl version` exist???
+
+### Upgrade Steps
+
+1. Stop all front-end nodes.
+
+    1. On Chef Infra Server 14.6.32 or later, put the server nodes into maintenance mode, which temporarily disables the API temporarily. if
+
+      ```bash
+      sudo chef-server-ctl maintenance on
+      ```
+
+    1. On Chef Infra Server 14.6.31 or earlier, stop all server nodes with the command:
+
+      ```bash
+      sudo chef-server-ctl stop
+      ```
+
+2. Identify the **leader** node:
+
+    Find **leader** node in the **Role** column in the output of the `cluster-status` subcommand. In this example, `backend-1` is the **leader** node, as indicated by its role in the **Role** column.
+
+    ```ruby
+    chef-backend-ctl cluster-status
+    ```
+
+    ```output
+    Name       IP              GUID                              Role      PG        ES
+    backend-1  192.168.33.215  dc0c6ea77a751f94037cd950e8451fa3  leader    leader    not_master
+    backend-2  192.168.33.216  008782c59d3628b6bb7f43556ac0c66c  follower  follower  not_master
+    backend-3  192.168.33.217  1af654172b1830927a571d9a5ba7965b  follower  follower  master
+    ```
+
+3. Install the new Chef Backend package on each of the server nodes in the cluster:
```

Comment:
Maybe like this? One of our customers mentioned this line being ambiguous during an actual upgrade

Install the new Chef Backend package on each of the backend server nodes in the cluster, including the current leader:

Reply:
Thank you for the gentle translation. I'm reasonably sure they didn't say `ambiguous`.

Reply:
What do you think of the plan for the upgrade content in this PR: https://github.com/chef/chef-server/pull/3147? I want to add a `Backend Upgrade` dropdown and separate this content by version.

Reply:
I don't have an immediate solution to making this clearer. I think we should come back to it.

Reply:
This could get a little confusing since we have 2 products that work together. Chef Infra Server and Chef Backend.
For a chef backend installation typically there would be atleast 4 servers:
3 backends - that install Chef Backend product
Atleast one frontend - that installs Chef Server product